### PR TITLE
Add Ubuntu 24.02 for QEMU

### PR DESCRIPTION
**Pull request by jrosser**
## Change description
Adds qemu image build for ubuntu 24.04

## Related issues
- Towards: https://github.com/kubernetes-sigs/image-builder/issues/1406

### CODE REVIEW

File: images/capi/packer/qemu/qemu-ubuntu-2404.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "boot_command_prefix": "c<wait>linux /casper/vmlinuz --- autoinstall ds='nocloud-net;s=http://{{ .HTTPIP }}:{{ .HTTPPort }}/24.04/'<enter><wait><wait><wait><wait>initrd /casper/initrd<wait><wait><enter><wait>boot<enter>",
+  "build_name": "ubuntu-2404",
+  "distribution_version": "2404",
+  "distro_name": "ubuntu",
+  "guest_os_type": "ubuntu-64",
+  "iso_checksum": "8762f7e74e4d64d72fceb5f70682e6b069932deedb4949c6975d0f0fe0a91be3",
+  "iso_checksum_type": "sha256",
+  "iso_url": "https://releases.ubuntu.com/releases/24.04/ubuntu-24.04-live-server-amd64.iso",
```
Comment (AverageMarcus) [inline images/capi/packer/qemu/qemu-ubuntu-2404.json:9]: This isn't ideal 😞  But right now there isn't an alternative. (see https://github.com/kubernetes-sigs/image-builder/issues/1334)

**Comment by jrosser** [comment]
This is failing `pull-azure-sigs` because of

```
azure-arm.sig-ubuntu-2204-cvm: RUNNING HANDLER [sysprep : Update debian grub] *********************************[0m
azure-arm.sig-ubuntu-2204-cvm: fatal: [default]: FAILED! => {"changed": true, "cmd": "update-grub\n", "delta": "0:00:00.003572", "end": "2024-06-17 07:03:22.913165", "msg": "non-zero return code", "rc": 127, "start": "2024-06-17 07:03:22.909593", "stderr": "/bin/sh: 1: update-grub: not found", "stderr_lines": ["/bin/sh: 1: update-grub: not found"], "stdout": "", "stdout_lines": []}
```

I suspect that these instance types don't use grub so this code should be made conditional, but I don't have a way to check that currently.

**Review by mboersma — APPROVED** [review]
/lgtm
/assign @AverageMarcus

**Review by AverageMarcus — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**